### ISSUE TITLE
.notNullable() on relationship identities exposes vulnerable information in error message

### ISSUE THREAD

**Issue by Jinjeleta** [postlight/lux#550]
Well, not entirely sure whether it could be counted as a bug, but I have found strange behavior in specific scenario, which exposes vulnerable information in response error message.

Case: we have two models: 
```js
class Question extends Model {
  static hasMany = {
    answers: {
      inverse: 'question'
    }
  }
```
and
```js
class Answer extends Model {
  static belongsTo = {
    question: {
      inverse: 'answers'
    }
  }
}
```
then we force to provide valid `qestion_id` upon creation of answer:
```js
  return schema.createTable('answers', table => {
    table.increments('id');

    table.text('body').notNullable();

    table.integer('question_id').notNullable().index(); // <= pay attention to this line
    table.foreign('question_id').references('question.id'); // <= pay attention to this line

    table.timestamps();

    table.index([
      'created_at',
      'updated_at'
    ]);
  });
```
We make request to create new `answer` without providing relationship information:
`http://localhost:4000/answers`
```js
{
  "data":{
    "type": "answers",
    "attributes": {
    	"body": "My new answer to your question!"
    }

  }
}
```
Response comes as:
```js
{
"errors": [
  {
"status": "500",
"title": "Internal Server Error",
"detail": "insert into "answers" ("body", "created_at", "question_id", "updated_at") values ('My new answer to your question!', '2016-12-01 19:42:56.828', NULL, '2016-12-01 19:42:56.828') - SQLITE_CONSTRAINT: NOT NULL constraint failed: answers.question_id"
}
],
"jsonapi": {
"version": "1.0"
}
}
```
However even when providing with valid question ID:
```js
{
  "data":{
  	"type": "answers",
    "attributes": {
    	"body": "My new answer to your question!"
    },
    "relationships": {
    	"question": { "data": { "type": "questions", "id": 1}}
    }
  }
}
```
it still does provide the same error message.

I have tested some more scenarios and it seems that `.notNullable()` is the real issue. You can conclude that yourself from log, when not using `.notNullable()` :
```
[2016-12-01T19:01:08.901Z] INSERT INTO "answers" ("body", "created_at", "question_id", "updated_at") VALUES ('MY NEW ANSWER TO YOUR QUESTION!', '2016-12-01 20:01:08.851', NULL, '2016-12-01 20:01:08.851')

------------------------------------------------------------------------------------------------------

[2016-12-01T19:01:08.963Z] SELECT "questions"."id" AS "id", "questions"."title" AS "title", "questions"."body" AS "body", "questions"."created_at" AS "createdAt", "questions"."updated_at" AS "updatedAt" FROM "questions" WHERE "questions"."id" = 1 LIMIT 1

------------------------------------------------------------------------------------------------------

[2016-12-01T19:01:09.009Z] UPDATE "answers" SET "question_id" = 1, "updated_at" = '2016-12-01 20:01:08.973' WHERE "id" = 2

------------------------------------------------------------------------------------------------------

[2016-12-01T19:01:09.017Z] Processed POST "/answers" from ::ffff:127.0.0.1 with 201 Created by AnswersController#create

```
I guess the logic goes like this: create answer with questioId as Null ==> check if given questionId can be found in questions table --> if yes, then update answer with valid questionId. In my opinion it's not most optimal way of doing it.

**Comment by zacharygolba** [comment]
Hey thanks for bringing this issue up!

> .notNullable() on relationship identities exposes vulnerable information in error message

The error message should be ambiguous when running in production mode `lux -e production` and not leak sensitive business logic to the client.

> I guess the logic goes like this: create answer with questioId as Null ==> check if given questionId can be found in questions table --> if yes, then update answer with valid questionId. In my opinion it's not most optimal way of doing it.

That is correct and I agree 100%. This is something we are fixing in #527. This will land in `v1.1`.

**Comment by zacharygolba** [comment]
Resolved by #527
